src/core/emergency-action.c

```diff
@@ -27,11 +39,23 @@ void emergency_action(
                 const char *reboot_arg,
                 int exit_status,
                 const char *reason) {
+        Unit *u;
 
         assert(m);
         assert(action >= 0);
         assert(action < _EMERGENCY_ACTION_MAX);
 
+        /* Is the special shutdown target active or queued? If so, we are in shutdown state */
+        u = manager_get_unit(m, SPECIAL_SHUTDOWN_TARGET);
+        if ((u && unit_active_or_pending(u))) {
```

Comment:
So... why is manager_state() not used here?

Reply:
I think changing to manager_state would be the cleanest way as follows:
        /* Is the special shutdown target active or queued? If so, we are in shutdown state */
        if ((manager_state(m) == MANAGER_STOPPING) &&
            ((action == EMERGENCY_ACTION_REBOOT) ||
             (action == EMERGENCY_ACTION_POWEROFF))) {
                log_info("EmergencyAction: Shutdown is already active Skipping %s request",
                         emergency_action_table[action]);
                return;
        }
However, this change would not make it easy to use log_unit_info() per the below question.